### PR TITLE
Feature/replace contract

### PR DESCRIPTION
**Pull request by ahrav**
## Summary by CodeRabbit

* **New Features**
  * Shared paging framework: non-empty pages, cursor-based continuation, page state, and capability flags for ordering/resumption/splitting; publicly exposed for connectors.

* **Bug Fixes / Validation**
  * Strict page validation with clear error reporting for empty, unsorted, duplicate, out-of-bounds, and inverted-range cases.

* **Tests**
  * Extensive unit and property tests covering normal and edge-case paging behaviors.

* **Documentation**
  * Updated connector docs describing the new paging vocabulary and validation.

### CODE REVIEW

File: crates/gossip-contracts/src/connector/common.rs [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    }
+}
+
+/// Optional paging behavior flags exposed by a connector family.
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
+pub struct PagingCapabilities {
+    pub ordered_keys: bool,
+    pub resumable: bool,
+    pub splittable: bool,
+}
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-contracts/src/connector/common.rs]: Declining this one. This project is pre-release with zero backwards-compatibility obligations and no downstream consumers — all changes are intentionally breaking, and callers are updated in the same commit. `#[non_exhaustive]` would add construction friction (`..Default::default()` required everywhere) with no practical benefit in this context.
